gprofiler/profilers/node.py

```diff
@@ -223,9 +223,10 @@ def get_node_processes() -> List[psutil.Process]:
     return pgrep_exe(r".*node[^/]*$")
 
 
-def generate_map_for_node_processes(processes: List[psutil.Process]) -> None:
+def generate_map_for_node_processes(processes: List[psutil.Process]) -> List[int]:
     """Iterates over all NodeJS processes, starts debugger for it, finds debugger URL,
     copies node-linux-perf module into process' namespace, loads module and starts it."""
+    pids_with_perf_map = []
```

Comment:
Keep `Process` objects, not PIDs. PIDs may be reused but a `Process` object can always be tested against the same process and you cannot possibly refer a different one